cpp/src/arrow/util/sse_util.h

```diff
@@ -22,31 +22,20 @@
 
 #include "arrow/util/macros.h"
 
-#ifdef ARROW_USE_SIMD
```

Comment:
Is `ARROW_USE_SIMD` still references in the codebase? If not, can you remove it from CMake?

Reply:
It's not used anymore in codebase.
It's used in cmake, acts as a central switch to turn on/off all simd flags such as ARROW_HAVE_SSE4_2.
Looks it's still useful? It's listed in benchmark doc. Code comment below:
` #Disable this option to exercise non-SIMD fallbacks`
`define_option(ARROW_USE_SIMD "Build with SIMD optimizations" ON)`

Reply:
Ah, ok. Let's keep it then.